### ISSUE TITLE
Rule Request: Prefer specific matcher over `XCTAssertEqual`/`XCTAssertNotEqual`

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1874]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

`XCTAssertFalse`, `XCTAssertTrue`,`XCTAssertNil`, `XCTAssertNotNil` should be used over `XCTAssertEqual`/`XCTAssertNotEqual`.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// triggers
XCTAssertEqual(foo, true)
XCTAssertEqual(foo, false)
XCTAssertNotEqual(foo, false)
XCTAssertEqual(foo, nil)
XCTAssertNotEqual(foo, nil)


// doesn't trigger
XCTAssertFalse(foo)
XCTAssertTrue(foo)
XCTAssertNil(foo)
XCTAssertNotNil(foo)
XCTAssertEqual(foo, 2)
XCTAssertNotEqual(foo, "false")
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Probably as a default rule, even though there's a risk of false positives.
